### DISCUSSION TITLE
In VSCode go to definition works only when the code compiles

### DISCUSSION THREAD

**Discussion by ylinik-dn** [rust-lang/rust-analyzer#14180] [Q&A] [answered]
Is there a reason it works this way? It is really annoying that even the smallest change breaks go to definition.

**Comment by flodiebold** [comment, marked as answer]
It doesn't usually work that way. Maybe there's a proc macro involved that fails to expand completely?

**Reply by ylinik-dn** [reply]
Seems like you are correct, not sure if I am misremembering things or a recent update to rust-analyzer fixed this.
Either way, thank you 🙏
